database/README.md

```diff
@@ -1,14 +1,42 @@
-This document explains the database that the orderbook api uses.
+Here we document the current state of the database. The history of these changes lives in the `sql` folder which contains all migrations. This document shows the schema and the purpose of the tables.
 
-We need to store orders persistently even if the orderbook service goes down. For this we use a PostgreSQL database. The schema can be found in `schema.sql`.
+Code that directly interfaces with the database through SQL queries lives in the `database`. This crate is often wrapped into higher level components by consumers.
 
-In addition to the orders, we also store trade events. This allows us to keep track of filled amounts.
+With a live database information for all tables can be retrieved with the `\d` command and information for a specific table with `\d MyTable`.
 
-To prevent users from spamming the database we should when receiving an order check that
-* the erc20 spending is approved
-* valid_to is not in the past and not more than <1 year> in the future
-* the order owner does not already have more than <1000> valid orders
+The database contains the following tables:
 
-This ensures that the amount of orders we have to keep track of per user is limited and that there is a real cost the user incurs before being able to submit orders because they need to have sent an ethereum transaction.
+### auction_participants
 
-Technically, this could still be attacked with a fake token that claims that approval has always been granted. This is more evidence to my belief that we should whitelist erc20 tokens that we will handle.
+   Column    |  Type  | Nullable | Default
+-------------|--------|----------|---------
+ auction_id  | bigint | not null |
+ participant | bytea  | not null |
+Indexes:
+- "auction_participants_pkey" PRIMARY KEY, btree (auction_id, participant)
+
+This table is used for [CIP-20](https://snapshot.org/#/cow.eth/proposal/0x2d3f9bd1ea72dca84b03e97dda3efc1f4a42a772c54bd2037e8b62e7d09a491f). It stores which solvers (identified by ethereum address) participated in which auctions (identified by auction id).
+
+TODO: CIP-20 makes no mention of a need to know about all participants. Why do we store this?
```

Comment:
open question

Reply:
Is that documented more officially? It isn't part of CIP-20.

Reply:
From the CIP:
```
The protocol commits to spend 20_000_000 COW tokens per year i.e. around 383307 COW per week on rewards. If at the end of an accounting period (currently: a week) the total amount of performance based rewards to be paid to solvers is smaller than the amount committed to by the protocol, the remaining sum is distributed to solver teams which consistently provide solutions. Those rewards will be distributed pro-rate to solvers based on the number of auctions they submitted a valid solution for over the accounting period.
```